File: api/src/main/java/com/datastrato/gravitino/Auditable.java
/*
 * Copyright 2023 Datastrato Pvt Ltd.
 * This software is licensed under the Apache License version 2.
 */
package com.datastrato.gravitino;

import com.datastrato.gravitino.annotation.Evolving;

/**
 * An auditable entity is an entity that has audit information associated with it. This audit
 * information is used to track changes to the entity.
 */
@Evolving
public interface Auditable {

  /** @return The audit information of the entity. */
  Audit auditInfo();
}

File: clients/client-python/gravitino/api/auditable.py
"""
Copyright 2024 Datastrato Pvt Ltd.
This software is licensed under the Apache License version 2.
"""

from abc import ABC, abstractmethod

from gravitino.api.audit import Audit


class Auditable(ABC):
    """
    An auditable entity is an entity that has audit information associated with it. This audit
    information is used to track changes to the entity.
    """

    @abstractmethod
    def audit_info(self) -> Audit:
        pass
